main.tf

```diff
@@ -26,7 +26,7 @@ resource "aws_kms_key" "this" {
 ################################################################################
 
 resource "aws_kms_external_key" "this" {
-  count = var.create && var.create_external ? 1 : 0
+  count = var.create && var.create_external && !var.create_replica && !var.create_replica_external ? 1 : 0
```

Comment:
hindsight is 20:20, but I should have named `var.create_external` -> `var.creat_external_key`